spec/controllers/dashboard_controller_spec.rb

```diff
@@ -461,20 +470,24 @@
     context 'changing tabs' do
       let(:group) { FactoryBot.create(:miq_group, :features => %w[everything]) }
       let(:user) { FactoryBot.create(:user_admin, :current_group => group, :miq_groups => [group]) }
+      let(:widget) { FactoryBot.create(:miq_widget) }
       let(:ws1) do
         FactoryBot.create(:miq_widget_set,
                           :name     => 'A',
-                          :owner_id => group.id,
-                          :set_data => {:col1 => [], :col2 => [], :col3 => []})
+                          :set_data => {:col1 => [widget.id],
```

Comment:
I noticed that a lot of these sets were incomplete which made me uncomfortable. That is where the smarter factory with encapsulation came into play. An alternative would be to put the setter into the actual model and treat these like fields (which is probably the correct schema/interface)